tests/performance-test/performance-test.sh

```diff
@@ -52,28 +124,59 @@ do
 done
 
 if [ "${TOOL}" = "tg" ]; then
-    MSGS=$((HOSTS * PLUGINS))
-    LENGTH_S=$((COUNT / INTERVAL))
-    CONFIG="\
-- metadata:
-    name: SAF Performance Test 1
-  spec:
-    value-lists: ${MSGS}
-    hosts: ${HOSTS}
-    plugins: ${PLUGINS}
-    interval: ${INTERVAL}
-    length: ${LENGTH_S}"
-
-    echo "${CONFIG}" > deploy/config/generated-test-configs.yml
-    cd deploy
-    export TG_CONFIGFILE=./config/generated-test-configs.yml
-    ./performance-test-tg.sh
-
+    echo "Collectd-tg not implemented. Try running with '-t tb' instead"
+    exit 1
 elif [ "${TOOL}" = "tb" ]; then
-    export COUNT HOSTS PLUGINS INTERVAL CONCURRENT
-    cd deploy
-    ./performance-test-tb.sh
-
+    :
 else
     usage
 fi
+
+get_sources
+create_grafana_datasources
+create_service_monitors
+post_dashboards
+
+STAGE="TARGET"
+while true; do
+    case $STAGE in
+        "TARGET")
+            TARGETS=`curl "http://$(oc get route prometheus -o jsonpath='{.spec.host}')/api/v1/targets"`
+            echo $TARGET
+            QDR=`echo $TARGETS | grep -o '"__meta_kubernetes_service_name":"qdr-white"'`
+            PROM=`echo $TARGETS | grep -o '"__meta_kubernetes_service_name":"prometheus-operated"'`
+
+            if [ -z "$QDR" ] && [ -z "$PROM" ]; then
+                echo "Waiting for new targets to be recognized by Prometheus Operator"
+                sleep 10
+            else
+                echo "Found new target endpoints. Creating performance test job"
+                export COUNT HOSTS PLUGINS INTERVAL CONCURRENT
+                cd deploy
+                ./performance-test-tb.sh
+                STAGE="TEST"
+            fi
+        ;;
+        "TEST")
+            estab=$(oc get pod -l job-name=saf-perftest-1-runner -o jsonpath='{.items[0].status.conditions[?(@.type=="ContainersReady")].status}') || true
+            if [ "${estab}" != "True" ]; then
+                echo "Waiting on SAF performance test pod creation..."
+                sleep 3
+            else
+                echo "SAF performance test pod established"
+                break
+            fi
+        ;;
+        *)
+            echo "Unrecognized state"
```

Comment:
Would it be appropriate to call `usage()` here?

Reply:
Potentially, though theoretically control flow should never arrive here and will definitely never arrive here because of user error. Its just a default to keep anything unexpected from happening if something like radiation happened to flip the bits in the state variable